### ISSUE TITLE
Bug - tv episodes with null aired date shoud be TBA unclickable

### ISSUE THREAD

**Issue by blackholeearth** [michaldrabik/showly-2.0#621]
Bug behavior: 
tv episodes with null aired_dates can be checked  by user

right/expected behavior:
tv episodes with null dates
 should be uncheckable. Should say TBA.

Note:
 in the previous versions,  expected behavior was the default behavior.

**Comment by michaldrabik** [comment]
Can you give an example of a TV Show that I could observe this on?

**Comment by blackholeearth** [comment]
Love on duty .  2022 series

https://trakt.tv/shows/love-on-duty

![XRecorder_07082022_112117](https://user-images.githubusercontent.com/7512105/183282216-1807093f-742a-4c11-9309-5ef1af11d205.jpg)

**Comment by michaldrabik** [comment]
In that case it's not a bug actually but a feature of the app.  Actually this series is an edge-case but still applies.

As episodes after 15th have their dates available the app assumes that missing dates is an API bug (which it is) because there is a "gap" in the season. The app does that becasue there are many many series with dates missing in the middle of the season for episodes which actually have been aired already.

Your case is actually unhandeled since its very rare. The episodes after the "gap" are all future dates so it should theoretically mark "gap" episodes as unable to check by defaut. I'll investigate that.

Does that makes sense?

**Comment by blackholeearth** [comment]
Yes.  Makes sense.

But incase of this serie.
people added 30 ep without knowing at which ep it will end.

This Serie made Final at 8 ep. 

İn showly ,  İ saw that ep 9 will be aired  in 3 days. 

So i went tmdb.  deleted dates of 9 to 15.( all False created episodes) 

At tmdb, İ also created report saying "ep8 is final please delete ep 9 to 33" . 
Who know when will they delete.!?

---------------
Tldr :::
 i will update episode's  dates in tmdb accordingly.

Episodes after final ep. İ will set them all to 1-1-2099 
So they won't false alert me in showly app.

Thanks.

**Comment by michaldrabik** [comment]
Yeah, if its updated Tmdb and Trakt will sync the data it will also be gone in the app. I will still add this case in the future update so that checkboxes will be disabled as described above.
